### PR TITLE
Spellcheck documentation

### LINKED ISSUE #555: Spell check the docs
Follow up task to #473.
In #473 only markdowlint is introduced.

### PR DESCRIPTION
**Pull request by Kielek**
Fixes #555 

Changes proposed in this pull request:

Recommended plugin for VS Code
Extended build pipeline for validating documentation
nuke/npm js step to execute validation step locally
Typo fixes!
Additional dictionaries.

Potential follow up
Execute cspell for whole code, not only documentation.

### CODE REVIEW

File: docs/design.md [hunk 1]
```diff
@@ -99,7 +99,7 @@ into the target process through a series of steps started by the CLR Profiler DL
 callback, the CLR Profiler DLL takes the following actions:
 
    - If the loaded module is in the set of modules for which there is bytecode instrumentation,
-   or if it's the first non-corelib module or not in one of the special cases,
+   or if it's the first non-core lib module or not in one of the special cases,
```
Comment (pellared) [inline docs/design.md:102]: I think corelib is a name used in .NET runtime

File: src/OpenTelemetry.AutoInstrumentation/Configuration/ConfigurationKeys.cs [hunk 2]
```diff
@@ -95,7 +95,7 @@ public class ConfigurationKeys
     /// <summary>
     /// Configuration key for enabling the flushing of telemetry data when an unhandled exception occurs.
     /// </summary>
-    public const string FlushOnUnhandledException = "OTEL_DOTNET_AUTO_FLUSH_ON_UNHANDLEDEXCEPTION";
+    public const string FlushOnUnhandledException = "OTEL_DOTNET_AUTO_FLUSH_ON_UNHANDLED_EXCEPTION";
```
Comment (pellared) [inline src/OpenTelemetry.AutoInstrumentation/Configuration/ConfigurationKeys.cs:98]: Update the changelog or revert

Comment (Kielek) [reply src/OpenTelemetry.AutoInstrumentation/Configuration/ConfigurationKeys.cs]: reverted

File: docs/design.md (hunk 1 above)
Comment (Kielek) [reply docs/design.md]: d6100a3a4df2fb79a4493831d4e83e92f5698de8

**Review by pellared — APPROVED** [review]

**Review by RassK — APPROVED** [review]

**Review by rajkumar-rangaraj — APPROVED** [review]

**Review by nrcventura — APPROVED** [review]

**Pull request merged**